app/views/admin/roles/_role_manager.haml

```diff
@@ -0,0 +1,63 @@
+              %label.btn.btn-secondary.btn-xs{ data: {'role-manager' => { target: 'administrativeFilter' }, action: 'click->role-manager#toggleAdmin' }}
+                %input{type: :radio, name: :roles, id: 'off-administrative-permissions', value: 'hide'} Hide
+            .ml-4
+              Administrative Permissions
+            .ml-auto
+              %i.icon-lock
+          %input.form-control.j-table__search.w-100{placeholder: 'Search permissions', data: { 'role-manager' => { target: 'searchInput' }, action: 'keyup->role-manager#searchPermissions'}}
+
+        .list-group.list-group-flush.role-names.border-right.border-dark
+          - @roles.each.with_index do |role, i|
+            - show = nil
+            - hide = nil
+            - show_class = nil
+            - hide_class = nil
+            - if i < 3
+              - show = :checked
+              - show_class = :active
+            - else
+              - hide = :checked
+              - hide_class = :active
+            %li.list-group-item.d-flex{style: "background-color: #{role.bg_color}"}
+              .btn-group.btn-group-toggle{data: {toggle: :buttons}, aria: {label: "Toggle #{role.name} visibility"}}
+                %label.btn.btn-secondary.btn-xs{class: show_class, data: { 'role-manager' => { target: 'roleToggle', 'role-value' => role.id }, action: 'click->role-manager#toggleColumn'}}
+                  %input{type: :radio, name: :roles, id: "on-#{role.name.downcase.gsub(/^a-z/, '-')}", checked: show, value: 'show'} Show
+                %label.btn.btn-secondary.btn-xs{class: hide_class, data: { 'role-manager' => { target: 'roleToggle', 'role-value' => role.id }, action: 'click->role-manager#toggleColumn'}}
+                  %input{type: :radio, name: :roles, id: "off-#{role.name.downcase.gsub(/^a-z/, '-')}", checked: hide, value: 'hide'} Hide
+              .ml-4{style: "color: #{role.fg_color}"}
+                = role.name
+      .role-wrapper.w-75{ data: { 'role-manager' => { target: 'inputWrapper', 'form-value' => path_for_patch }}}
+        = hidden_field_tag :authenticity_token, form_authenticity_token, class: 'j-authenticity-token'
+        - # cache permission calculation
+        - permissions = role_class.permissions_by_group
+        .d-flex
+          - @roles.each.with_index do |role, i|
+            - show_class = if i < 3 then nil else :hide end
+            .role-name-header.role-column.d-flex{style: "background-color: #{role.bg_color}", class: show_class, data: { 'role-manager' => { target: 'roleColumn', 'role-value' => role.id }}}
+              %h3{style: "color: #{role.fg_color}"}= role.name
+        .d-flex
+          - @roles.each.with_index do |role, i|
+            - show_class = if i < 3 then nil else :hide end
+            .role-column{class: show_class, data: { 'role-manager' => { target: 'roleColumn', 'role-value' => role.id }}}
+              = render 'admin/roles/role_column', role: role, permissions: permissions
```

Comment:
The column headers weren't aligning, I was able to fix with below
<img width="1581" alt="Screenshot 2024-02-29 at 8 44 20 AM" src="https://github.com/greenriver/hmis-warehouse/assets/5333982/871a2f42-0b96-447c-b728-fac80e759dcd">


```suggestion
        .d-flex
          - @roles.each.with_index do |role, i|
            - show_class = if i < 3 then nil else :hide end
            .role-name-header.role-column.d-flex{style: "background-color: #{role.bg_color}", class: show_class, data: { 'role-manager' => { target: 'roleColumn', 'role-value' => role.id }}}
              %h3{style: "color: #{role.fg_color}"}= role.name
        .d-flex
          - @roles.each.with_index do |role, i|
            - show_class = if i < 3 then nil else :hide end
            .role-column{class: show_class, data: { 'role-manager' => { target: 'roleColumn', 'role-value' => role.id }}}
              .role-name-header.role-column.d-flex{style: "background-color: #{role.bg_color}", class: show_class, data: { 'role-manager' => { target: 'roleColumn', 'role-value' => role.id }}}
                %h3{style: "color: #{role.fg_color}"}= role.name
              = render 'admin/roles/role_column', role: role, permissions: permissions
```